src/plugins/inspector/public/views/requests/components/details/req_code_viewer.tsx

```diff
@@ -6,68 +6,111 @@
+import { compressToEncodedURIComponent } from 'lz-string';
+import React, { MouseEvent, useCallback } from 'react';
+import { CodeEditor, useKibana } from '../../../../../../kibana_react/public';
 
 interface RequestCodeViewerProps {
+  indexPattern?: string;
   json: string;
 }
 
 const copyToClipboardLabel = i18n.translate('inspector.requests.copyToClipboardLabel', {
   defaultMessage: 'Copy to clipboard',
 });
 
+const openInDevToolsLabel = i18n.translate('inspector.requests.openInDevToolsLabel', {
+  defaultMessage: 'Open in Dev Tools',
+});
+
 /**
  * @internal
  */
-export const RequestCodeViewer = ({ json }: RequestCodeViewerProps) => (
-  <EuiFlexGroup
-    direction="column"
-    gutterSize="s"
-    wrap={false}
-    responsive={false}
-    className="insRequestCodeViewer"
-  >
-    <EuiFlexItem grow={false}>
-      <EuiSpacer size="s" />
-      <div className="eui-textRight">
-        <EuiCopy textToCopy={json}>
-          {(copy) => (
+export const RequestCodeViewer = ({ indexPattern, json }: RequestCodeViewerProps) => {
+  const { services } = useKibana();
+  const prepend = services.http?.basePath?.prepend;
+  const navigateToUrl = services.application?.navigateToUrl;
+
+  const devToolsDataUri = compressToEncodedURIComponent(`GET ${indexPattern}/_search\n${json}`);
+
+  const devToolsUrl = `/app/dev_tools#/console?load_from=data:text/plain,${devToolsDataUri}`;
+
+  const handleDevToolsLinkClick = useCallback(
+    (event: MouseEvent<HTMLAnchorElement>) => {
+      event.preventDefault();
+      if (navigateToUrl && prepend) {
+        navigateToUrl(prepend(devToolsUrl));
+      }
+    },
+    [devToolsUrl, navigateToUrl, prepend]
+  );
+
+  return (
+    <EuiFlexGroup
+      direction="column"
+      gutterSize="s"
+      wrap={false}
+      responsive={false}
+      className="insRequestCodeViewer"
+    >
+      <EuiFlexItem grow={false}>
+        <EuiSpacer size="s" />
+        <div className="eui-textRight">
+          <EuiCopy textToCopy={json}>
+            {(copy) => (
+              <EuiButtonEmpty
+                size="xs"
+                flush="right"
+                iconType="copyClipboard"
+                onClick={copy}
+                data-test-subj="inspectorRequestCopyClipboardButton"
+              >
+                {copyToClipboardLabel}
+              </EuiButtonEmpty>
+            )}
+          </EuiCopy>
+          {indexPattern && prepend && (
             <EuiButtonEmpty
               size="xs"
               flush="right"
-              iconType="copyClipboard"
-              onClick={copy}
-              data-test-subj="inspectorRequestCopyClipboardButton"
+              iconType="wrench"
+              href={prepend(devToolsUrl)}
+              onClick={handleDevToolsLinkClick}
+              data-test-subj="inspectorRequestOpenInDevToolsButton"
             >
-              {copyToClipboardLabel}
+              {openInDevToolsLabel}
             </EuiButtonEmpty>
           )}
```

Comment:
I think you only want to render the "Open in Dev Tools" button if the user is actually authorized to view the Dev Tools app. Otherwise would take some users to a 404 page.

You can check with the capabilities provided in `ApplicationStart` --
```
application.capabilities.dev_tools.show
```
Example: https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/dev_tools/public/application.tsx#L101-L107

Reply:
Minor nit: the `dev_tools` capability key isn't guaranteed to exist when you're checking outside of the `dev_tools` plugin itself, because the plugin may be disabled. You'll instead want something like:

```ts
   const canAccessDevTools = application.capabilities.dev_tools?.show ?? false;
```

Reply:
Added in d7f2cdac2a1.